tasklauncher/spring-cloud-starter-stream-sink-tasklauncher-cloudfoundry/src/test/java/org/springframework/cloud/stream/app/task/launcher/cloudfoundry/sink/TaskLauncherCloudfoundrySinkTests.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.task.launcher.cloudfoundry.sink;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.IntegrationTest;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryConnectionProperties;
+import org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryDeploymentProperties;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test Properties for the Task Launcher Cloud Foundry Sink.
+ *
+ * @author Glenn Renfro
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = TaskLauncherCloudfoundrySinkTests.TaskLauncherSinkApplication.class)
```

Comment:
We can discuss.  In short the master SCSt-Samples project uses older versions of the Reactor-Core and Netty.  In our case the CF-Client (used by the deployer) uses the latest Reactor* libraries.  Thus at compile time we don't need the Reactor* classes.  So we never had to declare them as dependencies.  
By adding the integration tests without the added dependencies, the old libraries would be used and thus the test fails.  To work around this weakness we had to add latest releases of these dependencies to this project .